platform.py

```
@@ -887,14 +896,131 @@ def _add_dynamic_options(self, board):
+        if not mcu_config:
+            return False
+        for tool_pkg in mcu_config["debug_tools"]:
+            pkg_dir = self.get_package_dir(tool_pkg)
+            if not pkg_dir:
+                continue
+            toolchain_arch = "xtensa-esp-elf" if mcu in MCU_TOOLCHAIN_CONFIG["xtensa"]["mcus"] else "riscv32-esp-elf"
+            gdb_path = Path(pkg_dir) / "bin" / f"{toolchain_arch}-gdb"
+            if not gdb_path.is_file():
+                continue
+            try:
+                result = subprocess.run(
```

Ruff (0.15.4)

[error] `subprocess` call: check for execution of untrusted input

(S603)